sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/LoadBalancerOutboundRule.java

```diff
@@ -0,0 +1,226 @@
+
+import java.util.List;
+import java.util.Map;
+
+/** An immutable client-side representation of an outbound rule. */
+public interface LoadBalancerOutboundRule
+    extends HasInnerModel<OutboundRuleInner>,
+    HasProtocol<LoadBalancerOutboundRuleProtocol>,
+    ChildResource<LoadBalancer> {
+
+    /** @return the associated frontends */
+    Map<String, LoadBalancerFrontend> frontends();
+
+    /** @return the associated backend */
+    LoadBalancerBackend backend();
+
+    /** @return the number of outbound ports to be used for NAT */
+    int allocatedOutboundPorts();
+
+    /** @return the provisioning state of the outbound rule resource */
+    ProvisioningState provisioningState();
+
+    /** @return the number of minutes before an idle connection is closed */
+    int idleTimeoutInMinutes();
+
+    /** @return if TCP reset is enabled */
+    boolean tcpResetEnabled();
+
+    /** @return outbound rule protocol */
+    LoadBalancerOutboundRuleProtocol protocol();
+
+    /** Grouping of outbound rule definition stages. */
+    interface DefinitionStages {
+        /**
+         * The first stage of the outbound rule definition.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface Blank<ParentT> extends LoadBalancerOutboundRule.DefinitionStages.WithProtocol<ParentT> {
+        }
+
+        /**
+         * The final stage of the outbound rule definition.
+         *
+         * <p>At this stage, any remaining optional settings can be specified, or the outbound rule definition can be
+         * attached to the parent load balancer definition.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithAttach<ParentT>
+            extends Attachable.InDefinition<ParentT>,
+            LoadBalancerOutboundRule.DefinitionStages.WithEnableTcpReset<ParentT>,
+            LoadBalancerOutboundRule.DefinitionStages.WithIdleTimeout<ParentT> {
+        }
+
+        /**
+         * The stage of an outbound rule definition allowing to specify the transport protocol.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithProtocol<ParentT>
+            extends HasProtocol.DefinitionStages.WithProtocol<LoadBalancerOutboundRule.DefinitionStages.WithBackend<ParentT>, LoadBalancerOutboundRuleProtocol> {
+        }
+
+        /**
+         * The stage of an outbound rule definition allowing to specify a backend pool for the rule to apply to.
+         *
+         * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+         */
+        interface WithBackend<ParentT>  {
+            LoadBalancerOutboundRule.DefinitionStages.WithFrontend<ParentT> fromBackend(String name);
```

Comment:
Does it need javadoc?

Reply:
Thanks for the reminder, will add javadoc